scripts/bert/dataset.py

```diff
@@ -28,28 +33,435 @@ class MRPCDataset(TSVDataset):
+        https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e
+    """
+    task_name = 'MNLI'
+    is_pair = True
+
+    def __init__(self,
+                 segment='dev_matched',
+                 root=os.path.join(
+                     os.getenv('GLUE_DIR', 'glue_data'), task_name)):  #pylint: disable=c0330
+        self._supported_segments = [
+            'dev_matched', 'dev_mismatched', 'test_matched', 'test_mismatched',
+            'diagnostic'
+        ]
+        assert segment in self._supported_segments, 'Unsupported segment: %s' % segment
+        path = os.path.join(root, '%s.tsv' % segment)
+        if segment in ['dev_matched', 'dev_mismatched']:
+            A_IDX, B_IDX, LABEL_IDX = 8, 9, 15
+            fields = [A_IDX, B_IDX, LABEL_IDX]
+        elif segment in ['test_matched', 'test_mismatched']:
+            A_IDX, B_IDX = 8, 9
+            fields = [A_IDX, B_IDX]
+        elif segment == 'diagnostic':
+            A_IDX, B_IDX = 1, 2
+            fields = [A_IDX, B_IDX]
+        super(MNLIDataset, self).__init__(
+            path, num_discard_samples=1, fields=fields)
+
+    @staticmethod
+    def get_labels():
+        """Get classification label ids of the dataset."""
+        return ['neutral', 'entailment', 'contradiction']
+
+    @staticmethod
+    def get_metric():
+        """Get metrics Accuracy"""
+        return Accuracy()
+
+
+@register(segment=['train', 'dev', 'test'])
+class WNLIDataset(GLUEDataset):
+    """Class for Winograd NLI task
+
+    Parameters
+    ----------
+    segment : str or list of str, default 'train'
+        Dataset segment. Options are 'train', 'val', 'test' or their combinations.
+    root : str, default '$GLUE_DIR/WNLI'
+        Path to the folder which stores the WNLI dataset.
+        The datset can be downloaded by the following script:
+        https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e
+    """
+    task_name = 'WNLI'
+    is_pair = True
+
+    def __init__(self,
+                 segment='train',
+                 root=os.path.join(
+                     os.getenv('GLUE_DIR', 'glue_data'), task_name)):
+        self._supported_segments = ['train', 'dev', 'test']
+        assert segment in self._supported_segments, 'Unsupported segment: %s' % segment
+        path = os.path.join(root, '%s.tsv' % segment)
+        if segment in ['train', 'dev']:
+            A_IDX, B_IDX, LABEL_IDX = 1, 2, 3
+            fields = [A_IDX, B_IDX, LABEL_IDX]
+        elif segment == 'test':
+            A_IDX, B_IDX = 1, 2
+            fields = [A_IDX, B_IDX]
+        super(WNLIDataset, self).__init__(
+            path, num_discard_samples=1, fields=fields)
+
+    @staticmethod
+    def get_labels():
+        """Get classification label ids of the dataset."""
+        return ['0', '1']
+
+    @staticmethod
+    def get_metric():
+        """Get metrics Accuracy"""
+        return Accuracy()
+
+
+def _truncate_seq_pair(tokens_a, tokens_b, max_length):
```

Comment:
This function is no longer needed (moved to https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/data/transforms.py#L1099)

Reply:
Thanks.